src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -151,7 +157,18 @@ def __init__(self):
 
     def execute(self, sql):
         logger.debug(f"[spark][execute] {sql}")
-        self._spark.sql(sql)
+        try:
+            immediate_response = self._spark.sql(sql)
+            status = immediate_response.status
+        except Exception as e:
+            status = StatementStatus(state=StatementState.FAILED, error=ServiceError(message=str(e)))
+
+        if status is None:
+            status = StatementStatus(state=StatementState.FAILED)
+        if status.state == StatementState.SUCCEEDED:
+            return immediate_response
+
+        self._raise_if_needed(status)
 
     def fetch(self, sql) -> Iterator[Row]:
```

Comment:
We also need to wrap exceptions for fetch()

Reply:
Done!